### PR TITLE
105803 deployment details page

### PR DESCRIPTION
**Pull request by oleterno-equinor**

### CODE REVIEW

File: src/components/alert/style.css [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   display: inline-block;
-  padding: var(--space-default-gap);
-  padding-left: calc(var(--space-default-gap) - var(--space-3));
+  padding: var(--eds_spacing_medium);
   position: relative;
 }
-
-.alert > svg {
-  margin-right: var(--space-default-gap);
+.alert.icon {
```
Comment (nilsgstrabo) [inline src/components/alert/style.css:7]: Create a reusable component for Icon+Content instead for hardcoding how icons should be styled in the Alert.

File: src/components/page-deployment/deployment-bread-crumb.js [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
-        { label: appName, to: routeWithParams(routes.app, { appName }) },
-        {
-          label: 'Deployments',
-          to: routeWithParams(routes.appDeployments, { appName }),
-        },
-        { label: smallDeploymentName(deploymentName) },
-      ]}
-    />
+    <Breadcrumbs>
+      <Breadcrumbs.Breadcrumb href={routeWithParams(routes.app, { appName })}>
```
Comment (nilsgstrabo) [inline src/components/page-deployment/deployment-bread-crumb.js:10]: react-router does not work with anchor elements rendered by Breeadcrumb. Remove href from Breadcrumb and add a NavLink component (see usage in the internal web-console breadcrumb component).

File: src/components/page-deployment/deployment-overview.js [hunk 3]
```diff
@@ -44,16 +44,11 @@ export class DeploymentOverview extends React.Component {
     }
 
     return (
-      <React.Fragment>
+      <div className="o-layout-constrained">
```
Comment (nilsgstrabo) [inline src/components/page-deployment/deployment-overview.js:47]: <div> with class o-layout-contrained should be removed. PageDeployment is a child to PageApplication, and PageApplication wraps all sub-components in a <div> with this class.

File: src/components/page-deployment/promote-deployment-action.js [hunk 4]
```diff
@@ -1,35 +1,28 @@
...
-                  fromEnvironment: deployment.environment,
-                }
-              )}
-            >
-              Promote deployment…
-            </LinkButton>
-          </ActionsPage>
+    <div>
+      <Button
+        href={routeWithParams(
```
Comment (nilsgstrabo) [inline src/components/page-deployment/promote-deployment-action.js:12]: <Button> has the same navigation issue as Breadcrumb; full page reload.
Consider wrapping EDS Button component into existing LinkButton in web-console.

File: src/components/page-deployment/promote-deployment-action.js [hunk 5]
```diff
@@ -1,35 +1,28 @@
...
+          routes.appJobNew,
+          { appName },
+          {
+            pipeline: 'promote',
+            deploymentName: deploymentName,
+            fromEnvironment: deployment.environment,
+          }
         )}
-    </span>
+        className="button"
```
Comment (nilsgstrabo) [inline src/components/page-deployment/promote-deployment-action.js:21]: remove this class and all styling in elements.links.css

File: src/style/elements.links.css [hunk 6]
```diff
@@ -7,7 +7,15 @@ a:hover {
...
+}
+
 a:focus {
   outline: none;
   text-decoration: underline;
 }
+
+a.button:focus {
+  text-decoration: none;
+}
```
Comment (nilsgstrabo) [inline src/style/elements.links.css:21]: This file should be removed. It overrides EDS component styling.

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
Some changes required

File: src/components/alert/style.css (hunk 1 above)
Comment (oleterno-equinor) [reply src/components/alert/style.css:7]: Let's make that a separate task.

File: src/components/page-deployment/promote-deployment-action.js (hunk 4 above)
Comment (oleterno-equinor) [reply src/components/page-deployment/promote-deployment-action.js]: wrapped EDS Button inside a NavLink instead.

**Review by nilsgstrabo — APPROVED** [review]
Approved. Agreed to fix issue with Breadcrumb in a separate task.

**Pull request merged**